### PR TITLE
Remove redundant getVotes call in Governor

### LINKED ISSUE #4921: Potentially redundant getVotes in Governor propose function
The following block of code checks if a proposer has enough voting power. 
However, if the threshold is zero, this check is redundant and does an extra `getVotes`, which can be quite expensive.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/01ef448981be9d20ca85f2faf6ebdf591ce409f3/contracts/governance/Governor.sol#L288-L291

This block can be replaced with:

```solidity
 // check proposal threshold 
 uint256 votesThreshold = proposalThreshold(); 
 if (votesThreshold > 0) {
   uint256 proposerVotes = getVotes(proposer, clock() - 1); 
   if (proposerVotes < votesThreshold) { 
```

### PR DESCRIPTION
**Pull request by aman7100**
Fixes #4921 

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Comment by ernestognw** [comment]
I guess this is for #4921, but please help us by writing a quick explanation and also linting the code at least.

**Comment by ernestognw** [comment]
Closed by accident

File: .changeset/healthy-timers-collect.md
Comment (ernestognw) [inline .changeset/healthy-timers-collect.md:5]: I'm not sure if this requires a changeset, but if it does, this phrasing doesn't tell anything. Would you mind updating the changeset with a concrete explanation of what's being solved?

Comment (ernestognw) [reply .changeset/healthy-timers-collect.md:5]: We can disable the check if needed.

Please refer to [the GUIDELINES](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7bd2b2aaf68c21277097166a9a51eb72ae239b34/GUIDELINES.md) before we review further.

> All changes to the core codebase (excluding tests, auxiliary scripts, etc.) must be documented in a changelog, except for purely cosmetic or documentation changes.

Also review our [CHANGELOG](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6ba452dea4258afe77726293435f10baf2bed265/CHANGELOG.md) for a reference of how to write a proper changeset.

Comment (aman7100) [reply .changeset/healthy-timers-collect.md:5]: Is it important to add a changeset for a small issue like this?

File: contracts/governance/Governor.sol
Comment (Amxx) [inline contracts/governance/Governor.sol:289]:
```suggestion
```

File: .changeset/healthy-timers-collect.md
Comment (Amxx) [reply .changeset/healthy-timers-collect.md:5]: I think this PR should not have a changelog entry.

Comment (ernestognw) [reply .changeset/healthy-timers-collect.md]: > I think this PR should not have a changelog entry.

Pretty much agree.

**Comment by ernestognw** [comment]
I removed `package.json` since it includes the version bump from contracts 5.0.1 to 5.0.2. I think this PR should focus only on the Governor change.

**Comment by aman7100** [comment]
Can you state what should i do next?

**Comment by ernestognw** [comment]
> Can you state what should i do next?

I'm approving if tests pass. Let's wait for that and we'll need a second approval from @Amxx

**Comment by Amxx** [comment]
I'm sure the tests will pass. What I'm curious about is the gas cost change when the threshold is not zero

**Review by Amxx — APPROVED** [review]

**Pull request merged**
